File: sample/src/main/java/library/minimize/com/chronometer/MainActivity.java
package library.minimize.com.chronometer;

import android.content.SharedPreferences;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import library.minimize.com.chronometer.databinding.ActivityMainBinding;
import library.minimize.com.chronometerpersist.ChronometerPersist;

public class MainActivity extends AppCompatActivity {
  ActivityMainBinding mBinding;
  ChronometerPersist chronometerPersist;

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
    SharedPreferences sharedPreferences = getSharedPreferences("ChronometerSample", MODE_PRIVATE);
    chronometerPersist =
        ChronometerPersist.getInstance(mBinding.chronometerView, sharedPreferences);
    //chronometerPersist.hourFormat(true);
    mBinding.start.setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        if (chronometerPersist.isRunning()) {
          mBinding.start.setText(R.string.resume);
          chronometerPersist.pauseChronometer();
        } else {
          mBinding.start.setText(R.string.pause);
          chronometerPersist.startChronometer();
        }
      }
    });

    mBinding.stop.setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        mBinding.start.setText(R.string.start);
        chronometerPersist.stopChronometer();
      }
    });
  }

  @Override protected void onResume() {
    super.onResume();
    chronometerPersist.resumeState();
    if (chronometerPersist.isRunning()) {
      mBinding.start.setText(R.string.pause);
    } else if (chronometerPersist.isPaused()) {
      mBinding.start.setText(R.string.resume);
    }
  }
}

File: sample/src/main/java/library/minimize/com/chronometer/MainActivity.kt
package library.minimize.com.chronometer

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import library.minimize.com.chronometer.databinding.ActivityMainBinding
import library.minimize.com.chronometerpersist.ChronometerPersist

class MainActivity : AppCompatActivity() {
    private var mBinding: ActivityMainBinding? = null
    private var chronometerPersist: ChronometerPersist? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        val mBinding = mBinding ?: return

        val sharedPreferences = getSharedPreferences("ChronometerSample", Context.MODE_PRIVATE)
        chronometerPersist = ChronometerPersist.getInstance(
            chronometer = mBinding.chronometerView,
            identifier = "mainChronometer",
            sharedPreferences = sharedPreferences
        )
        val chronometerPersist = chronometerPersist ?: return
        //chronometerPersist.hourFormat(true);
        mBinding.start.setOnClickListener {
            if (chronometerPersist.isRunning) {
                mBinding.start.setText(R.string.resume)
                chronometerPersist.pauseChronometer()
            } else {
                mBinding.start.setText(R.string.pause)
                chronometerPersist.startChronometer()
            }
        }

        mBinding.checkBoxHourFormat.setOnCheckedChangeListener { _, isChecked ->
            chronometerPersist.hourFormat(
                isChecked
            )
        }

        mBinding.stop.setOnClickListener {
            mBinding.start.setText(R.string.start)
            chronometerPersist.stopChronometer()
        }
    }

    override fun onResume() {
        super.onResume()
        val chronometerPersist = chronometerPersist ?: return
        val mBinding = mBinding ?: return

        chronometerPersist.resumeState()
        if (chronometerPersist.isRunning) {
            mBinding.start.setText(R.string.pause)
        } else if (chronometerPersist.isPaused) {
            mBinding.start.setText(R.string.resume)
        }
    }
}
